Sources/Harmony/Data/Query/NetworkQuery.swift

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2022 Mobile Jazz SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import Alamofire
+
+open class NetworkQuery: KeyQuery {
+
+    public enum Method: Equatable {
+        
+        case get
+        case delete
+        case put(type: ContentType?)
+        case post(type: ContentType?)
+
+        public static func ==(lhs: NetworkQuery.Method, rhs: NetworkQuery.Method) -> Bool {
+            switch (lhs, rhs) {
+            case (.get, .get):
+                return true
+            case(.delete, .delete):
+                return true
+            case (.put, .put):
+                return true
+            case (.post, .post):
+                return true
+            default:
+                return false
+            }
+        }
+        
+        public func with(contentType newContentType: ContentType?) -> Self {
+            if case .put = self {
+                return .put(type: newContentType)
+            } else if case .post = self {
+                return .post(type: newContentType)
+            } else {
+                return self
```

Comment:
if we do `.with(contentType:...)` on a get or delete, will not complain. Is this correct? should we crash instead? print a warning?

Reply:
I was intentionally not complaining, my reasoning being that changing the content type on a method that does not support a content type should still return the original method (delete or get). 

I can make it throw, may be it's clearer that way? But it was intentional.

Reply:
Done.